### PR TITLE
[Bug] Fix custom map style input

### PR DESCRIPTION
**Pull request by akre54**
Fix for #2560 and #2458: custom map style urls aren't being loaded.

Quoting [from #2560](https://github.com/keplergl/kepler.gl/issues/2560#issuecomment-2065578506):

Looks like the issue (or one issue) is that the `confirmButton` in the AddMapStyle modal is always disabled. It's checking for `mapStyle.inputStyle.style` which is always null since the input `onChange` listener only sets a `url` and not a `style` property. Also `addCustomMapStyleUpdater` is checking for `state.inputStyle.id` which is also null. Setting any id from the `onChange` handler seems to fix the issue.

The icon is also broken. The getStyleImageIcon called from inputMapStyleUpdater is returning a Mapbox url even for non-mapbox basemaps. For instance, inputting the example url "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json" gives me "https://api.mapbox.com/styles/v1/https://basemaps.cartocdn.com/gl/positron-gl-style/style.json/static/-122.3391,37.7922,9,0,0/400x300?access_token={snip}&logo=false&attribution=false". I'm not sure what the best fix for that would be.

### CODE REVIEW

File: src/cloud-providers/package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "umd"
   ],
   "dependencies": {
+    "@auth0/auth0-spa-js": "^2.1.2",
```
Comment (ibgreen) [inline src/cloud-providers/package.json:33]: Is it possible to separate this PR from the dependency fix so we can discuss them each on their merits?

Comment (akre54) [reply src/cloud-providers/package.json]: Done. I had to branch off of the other one to get the fix

**Comment by ibgreen** [comment]
> The icon is also broken. The getStyleImageIcon called from inputMapStyleUpdater is returning a Mapbox url even for non-mapbox basemaps. For instance, inputting the example url "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json" gives me "[https://api.mapbox.com/styles/v1/https://basemaps.cartocdn.com/gl/positron-gl-style/style.json/static/-122.3391,37.7922,9,0,0/400x300?access_token={snip}&logo=false&attribution=false](https://api.mapbox.com/styles/v1/https://basemaps.cartocdn.com/gl/positron-gl-style/style.json/static/-122.3391,37.7922,9,0,0/400x300?access_token=%7Bsnip%7D&logo=false&attribution=false)". I'm not sure what the best fix for that would be.

Looking at your URL it seems the style is expected to be a name not a URL (see the double https in `https://api.mapbox.com/styles/v1/https://basemaps.cartocdn.com/gl/positron-gl`).

So while this PR may unbreak some thing, it is probably not a complete solution.

@ilyabo @macrigiuseppe Any thoughts?

**Comment by akre54** [comment]
I can confirm that the main issue is still broken with just a mapbox url scheme:


https://github.com/keplergl/kepler.gl/assets/931368/23aa7a69-2186-40f8-a31a-37af3aac941f

**Comment by akre54** [comment]
With this fix, the `mapbox://` url scheme still doesn't work. Looks like the only two places that check for a mapbox url scheme are `isStyleUsingMapboxTiles` (called to determine attribution in `map-container.tsx`) and `getStyleDownloadUrl` (called from `getLoadMapStyleTasks`)


https://github.com/keplergl/kepler.gl/assets/931368/faf578e9-c335-4c82-ac4e-c9b093d16178

**Comment by ibgreen** [comment]
I don't think that reverting to mapbox is what we want to do. Supporting both would be an option, though that is a feature that someone needs to implement.

One option is to remove the map style support and the dialog.

How important is this feature? Are we breaking a important workflows here?

**Comment by akre54** [comment]
Would it be possible to simply rewrite the urls? Either in Kepler or in Maplibre? This feels like a pretty useful feature

**Comment by akre54** [comment]
Or at the very least include a warning message if you try to use a `mapbox://` url scheme? This feels unexpected

**Comment by ilyabo** [comment]
I suspect that newer mapbox styles would not work correctly with maplibre. I have experienced issues when trying to use newer styles with mapbox@1.

**Comment by akre54** [comment]
I can look into the icon import. It feels odd that the importer would assume that all urls come from mapbox, but it's not clear to me that Carto, for instance, returns a unique icon for all of its styles, or that every other style provider is required to return one as well. It seems to me like this would be better if it just used a generic icon, or black square.

I do think it's fine for `mapbox://` urls to be broken so long as basic style.jsons still work. What do you think?

**Comment by chrisgervang** [comment]
It doesn't really make sense to continue supporting Mapbox styles (or their URLs) unless the actual Mapbox library is added as an alternative option, right?

**Review by ibgreen — APPROVED** [review]

**Comment by ibgreen** [comment]
Apologies that this is taking time, the basemap support in kepler clearly lacks a decisive owner right now. Together we all have parts of the answer, but the risk is that this thread could go on for a long time unless we decide on something. Following up in the issue.

**Comment by akre54** [comment]
Yeah totally agree. I don't want to bike shed this any more than necessary. I agree there needs to be a better default for the icon. If I come up a patch to make this black or a basic color or something would that be acceptable?

**Comment by ibgreen** [comment]
>  come up a patch to make this black or a basic color or something would that be acceptable?

Yes let's land something reasonable.

**Comment by birkskyum** [comment]
> It doesn't really make sense to continue supporting Mapbox styles (or their URLs) unless the actual Mapbox library is added as an alternative option, right?

I think this is right. The reason mapbox:// was removed from maplibre was not only a matter of compatibility, but also a license compliance concern. This [change in the style spec license](https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/LICENSE.txt) saying:

> The software and files in this repository (collectively, "Software") are
licensed under the Mapbox TOS for use only with the relevant Mapbox product(s)
listed at www.mapbox.com/pricing.

Looking at the www.mapbox.com/pricing etc. it seemed unsafe to interpret that in any other way than "mapbox styles are only allowed to be rendered with a mapbox renderer".

**Comment by akre54** [comment]
I added a temporary fix using the `NO_BASEMAP_ICON` for custom map styles and removing the logic for `isValidStyleUrl` which checked for mapbox styles. I'm sure this can be changed in the future but for now this fixes the major issues.

File: src/utils/src/index.ts [hunk 2]
```diff
@@ -70,7 +70,6 @@ export {
   getDefaultLayerGroupVisibility,
   editTopMapStyle,
   editBottomMapStyle,
-  isValidStyleUrl,
```
Comment (akre54) [inline src/utils/src/index.ts:73]: This import was unused

**Comment by birkskyum** [comment]
seems like ci is failing on the "should set the inputStyle" test

**Comment by akre54** [comment]
@birkskyum oddly tests are passing for me locally. Not sure what's happening on CI?

**Comment by birkskyum** [comment]
Weird,. Maybe the ci has different node version, or OS. What's your environment like?

**Comment by akre54** [comment]
I'm on Mac node v20.9.0. That test is a unit test that isn't using a Chrome instance, so I'm not sure why it would be massively different. Are you able to run it locally?

**Comment by birkskyum** [comment]
I see it fail locally too, with `yarn test-node-debug` and node 18

File: test/node/reducers/map-style-test.js [hunk 3]
```diff
@@ -836,7 +836,7 @@ test('#mapStyleReducer -> EDIT_CUSTOM_MAP_STYLE', t => {
 
   t.deepEqual(
     nextState.inputStyle,
-    {...inputMapStyleToEdit, error: false, isValid: true, style: null, uploadedFile: null},
+    {...inputMapStyleToEdit, error: false, isValid: false, style: null, uploadedFile: null, url: },
```
Comment (birkskyum) [inline test/node/reducers/map-style-test.js:839]: The `url:` syntax looks a bit strange to me

**Pull request merged**